test/e2e/utils/kubernetes.go

```diff
@@ -638,7 +638,16 @@ func (k TestKubernetes) WaitForPods(required int, timeout time.Duration, listOps
 	return podList
 }
 
-func (k TestKubernetes) WaitForInfinispanConditionWithTimeout(name, namespace string, condition ispnv1.ConditionType, timeout time.Duration) *ispnv1.Infinispan {
+func (k TestKubernetes) WaitForInfinispanConditionWithTimeout(name, namespace string, condition ispnv1.ConditionType, timeout time.Duration, value ...string) *ispnv1.Infinispan {
+	if len(value) > 0 && value[0] == string(metav1.ConditionFalse) {
+		return k.WaitForInfinispanStateWithTimeout(name, namespace, timeout, func(i *ispnv1.Infinispan) bool {
+			if i.IsConditionFalse(condition) {
+				log.Info("infinispan condition met", "condition", condition, "status", metav1.ConditionTrue)
+				return true
+			}
+			return false
+		})
+	}
```

Comment:
Instead of having the `if` logic in this function we can just add the logic directly to the `WaitForInfinispanConditionFalse` function:

```go
func (k TestKubernetes) WaitForInfinispanConditionFalse(name, namespace string, condition ispnv1.ConditionType) *ispnv1.Infinispan {
	return k.WaitForInfinispanStateWithTimeout(name, namespace, timeout, func(i *ispnv1.Infinispan) bool {
		if i.IsConditionFalse(condition) {
			log.Info("infinispan condition met", "condition", condition, "status", metav1.ConditionFalse)
			return true
		}
		return false
	})
}
```